### ISSUE TITLE
Assess computational benefit of a single temporary MultiFab in the diagnostic class

### ISSUE THREAD

**Issue by RevathiJambunathan** [ECP-WarpX/WarpX#922]
In the recently reformatted diagnostic implementation, temporary MultiFabs are created and used for computing divE, divB, particle_per_cell, particles_per_grid, in their respective functors (Source/Diagnostics/ComputDiagFunctors/*.cpp). 
Instead of defining temporary multifabs for each of the above mentioned functors, we need to assess if there is some benefit in allocating a single temporary multifab variable, `mf_tmp` just once in Diagnostics::ComputeAndPack, and passing it as a pointer in the operator() function?
If it does improve the performance, then, we will adopt this strategy of using a single tmp_mf.

**Comment by ax3l** [comment]
Awesome suggestion! I think independent of performance, this could lower our memory footprint as well and allow us to run larger simulations on the same GPU hardware, what do you think?

**Comment by RevathiJambunathan** [comment]
yes, I think we allocate-deallocate to limit memory peak, each time an operator() for the functors is called in diagnostics.
